nbformat/json_compat.py

```diff
@@ -47,6 +48,9 @@ def validate(self, data):
             raise ValidationError(error.message, schema_path=error.path)
 
     def iter_errors(self, data, schema=None):
+        if schema is not None:
+            return self._default_validator.iter_errors(data, schema)
+
```

Comment:
Because of the check above, we know `schema` is `None` now so the code below can be:

```
errors = []
validate_func = self._validator
```

Reply:
Good catch Fixing!